docs/types/resolvers-self.md

```diff
@@ -0,0 +1,218 @@
+---
+Title: Accessing parent's data in resolvers
+---
+
+# Accessing parent's data in resolvers
+
+It is quite common to want to be able to access the data from the field's parent
+in a resolver. For example let's say that we want to define a `fullName` field
+on our `User`. This would be our code:
+
+```python+schema
+import strawberry
+
+
+@strawberry.type
+class User:
+    first_name: str
+    last_name: str
+    full_name: str
+---
+type User {
+    firstName: String!
+    lastName: String!
+    fullName: String!
+}
+```
+
+In this case `full_name` will need to access the `first_name` and `last_name`
+fields, and depending on whether we define the resolver as a function or as a
+method, we'll have a few options! Let's start with the defining a resolver as a
+function.
+
+## Accessing parent's data in function resolvers
+
+```python
+import strawberry
+
+
+def get_full_name() -> str: ...
+
+
+@strawberry.type
+class User:
+    first_name: str
+    last_name: str
+    full_name: str = strawberry.field(resolver=get_full_name)
+```
+
+Our resolver is a function with no arguments, in order to tell Strawberry to
+pass us the parent of the field, we need to add a new argument with type
+`strawberry.Parent[ParentType]`, like so:
+
+```python
+def get_full_name(parent: strawberry.Parent[User]) -> str:
+    return f"{parent.first_name} {parent.last_name}"
+```
+
+`strawberry.Parent` tells Strawberry to pass the parent value of the field, in
+this case it would be the `User`.
+
+> **Note:** `strawberry.Parent` accept a type argument, which will then be used
+> by your type checker to check your code!
+
+### Using root
+
+Historically Strawberry only supported passing the parent value by adding a
+parameter called `root`:
+
+```python
+def get_full_name(root: User) -> str:
+    return f"{root.first_name} {root.last_name}"
+```
+
+This is still supported, but we recommend using `strawberry.Parent`, since it
+follows Strawberry's philosofy of using type annotations. Also, with
+`strawberry.Parent` your argument can be named anything, for example this will
+still work:
+
+```python
+def get_full_name(user: strawberry.Parent[User]) -> str:
+    return f"{user.first_name} {user.last_name}"
+```
+
+## Accessing parent's data in a method resolver
+
+Both method also work when defining a method resolver, so we can still use
```

Comment:
`Both methods` maybe?

But also try to avoid `method` to describe the "method of doing something", as it can be ambiguous with an object method as defined here